src/test/validation_block_tests.cpp

```diff
@@ -340,4 +340,30 @@ BOOST_AUTO_TEST_CASE(mempool_locks_reorg)
         rpc_thread.join();
     }
 }
+
+BOOST_AUTO_TEST_CASE(witness_commitment_index)
+{
+    CScript pubKey;
+    pubKey << 1 << OP_TRUE;
+    auto ptemplate = BlockAssembler(*m_node.mempool, Params()).CreateNewBlock(pubKey);
+    auto pblock = std::make_shared<CBlock>(ptemplate->block);
```

Comment:
Doesn't need to be a shared pointer, since you are not passing it around?

```suggestion
    CBlock block = ptemplate->block;
```